scripts/data/src/org/akvo/gae/remoteapi/CountFormInstances.java

```diff
@@ -0,0 +1,183 @@
+                    surveyCounts.get(survey),
+                    surveyDates.get(survey),
+                    parentName))
+            .append("\n");
+      }
+    }
+  }
+
+  private void fetchSurveyGroups(DatastoreService ds) {
+
+    final Query q = new Query("SurveyGroup");
+    final PreparedQuery pq = ds.prepare(q);
+
+    for (Entity g : pq.asIterable(FetchOptions.Builder.withChunkSize(500))) {
+
+      Long surveyGroupId = g.getKey().getId();
+      Long parentId = (Long) g.getProperty("parentId");
+      String type = (String) g.getProperty("projectType");
+      String name = (String) g.getProperty("name");
+      if (parentId == null) {
+      } else {
+        sgParents.put(surveyGroupId, parentId);
+        sgName.put(surveyGroupId, name);
+        sgType.put(surveyGroupId, type);
+      }
+    }
+  }
+
+  private void fetchSurveys(DatastoreService ds) {
+
+    final Query survey_q = new Query("Survey");
+    final PreparedQuery survey_pq = ds.prepare(survey_q);
+
+    for (Entity s : survey_pq.asIterable(FetchOptions.Builder.withChunkSize(500))) {
+
+      Long surveyId = s.getKey().getId();
+      String surveyName = (String) s.getProperty("name");
+      Long surveyGroup = (Long) s.getProperty("surveyGroupId");
+      if (surveyGroup != null) {
+        // FILTER THE SURVEY ID
+        Filter fsi = new FilterPredicate("surveyId", FilterOperator.EQUAL, surveyId);
+        // FILTER ONLY FOR THE LAST SEMESTER
+        Date dt = Date.from(ZonedDateTime.now().minusMonths(6).toInstant());
+        Filter fdt =
+            new FilterPredicate("createdDateTime", FilterOperator.GREATER_THAN_OR_EQUAL, dt);
+        Filter fmrg = CompositeFilterOperator.and(fsi, fdt);
+        /* END FILTER
+        / We should consider that not using setKeysOnly will return full entities
+        / Documentation :
+        / https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/Query.html#setKeysOnly
+        */
+        Query si =
+            new Query("SurveyInstance")
+                .setFilter(fmrg)
+                .addSort("createdDateTime", SortDirection.ASCENDING);
+        //  Query si = new Query("SurveyInstance").setFilter(fmrg).setKeysOnly();
+        long count = 0;
+        Date surveyDate = null;
+        for (@SuppressWarnings("unused")
+        Entity sie : ds.prepare(si).asIterable(FetchOptions.Builder.withChunkSize(500))) {
+          surveyDate = (Date) sie.getProperty("createdDateTime");
+          count++;
+        }
+        if (count > 0) {
+          DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+          String strDate = dateFormat.format(surveyDate);
+          surveyCounts.put(surveyId, count);
+          surveyNames.put(surveyId, surveyName);
+          surveyDates.put(surveyId, strDate);
+          surveyParents.put(surveyId, surveyGroup);
+          surveyToGroups.put(surveyId, new ArrayList<Long>());
+        }
+      }
+    }
+  }
```

Comment:
My suggestion here would be things.

Second I would use the `setKeysOnly()` for the SurveyInstance query. Since you only want to count you do not need any other info from the Survey instance other than how many are there. If you do it the way it is now, it will increase the number of reads and that could increase our monthly bill for the instances considerably.  setKeysOnly does not count towards the reads/cost.

Reply:
Let's wait for @joycarpediem to answer.

Reply:
@joycarpediem confirmed that for now, we don't need the last updated data. I'm going to comment on the line then merge @muloem